azurelinuxagent/ga/exthandlers.py

```diff
@@ -436,6 +436,9 @@ def handle_ext_handler(self, ext_handler, etag):
             else:
                 message = u"Unknown ext handler state:{0}".format(state)
                 raise ExtensionError(message)
+        except ExtensionUpdateError as e:
+            # Only setting the handler status here as the error has already been reported from the old version
+            ext_handler_i.set_handler_status(message=ustr(e), code=e.code)
```

Comment:
this works fine now, but if we ever add/change logic in handle_ext_handler_error (which currently is the central place to handle errors) we may miss this one... consider adding a flag to handle_ext_handler_error to do/not do the logging and call it from here as well.

Reply:
Makes sense, I'll modify the code